pkg/dao/templates.go

```diff
@@ -181,7 +181,10 @@ func (t templateDaoImpl) Update(ctx context.Context, orgID string, uuid string,
 	})
 
 	if err != nil {
-		return resp, fmt.Errorf("could not update template %w", err)
+		if err == gorm.ErrRecordNotFound {
+			return resp, &ce.DaoError{NotFound: true, Message: "Could not find template with UUID " + uuid}
+		}
+		return resp, t.DBToApiError(err)
```

Comment:
if i tried to PUT or PATCH a template with an invalid uuid i would see this, which i didn't think was expected:
`error="error: code=500, title=Error updating template, detail=could not update template Could not find template with UUID 0810dee5-531e-4cff-9439-9c1bdec89b3 \n"`

Reply:
Ah I see. I think what we should change here instead is to remove the wrapping error i.e ["Could not update tempate"](https://github.com/content-services/content-sources-backend/blob/main/pkg/dao/templates.go#L184). Then the error should pass up from the dao properly.

Reply:
It would be nice if we didn't have to worry about wrapping it, and it still found the right error underneath the wrapped error, but I don't how big of a change that would be.

Reply:
got it. if i remove both the update error (could not update template) and the fetch error right after it (could not fetch template), then i see the expected 404. so this should be ok since those are handled in the fetch method that's called by both update and Fetch?

Reply:
yep!

Reply:
I still agree with removing those extra wrapping errors because I think they were extraneous. But, I did figure out how to get the DaoError even if it's wrapped. We change this function: https://github.com/content-services/content-sources-backend/blob/main/pkg/errors/handler_error.go#L93-L106

to this
```
// HttpCodeForDaoError returns http code for corresponding dao error
func HttpCodeForDaoError(err error) int {
	var daoError *DaoError
	ok := errors.As(err, &daoError)
	if ok {
		if daoError.NotFound {
			return http.StatusNotFound
		} else if daoError.BadValidation {
			return http.StatusBadRequest
		} else if daoError.AlreadyExists {
			return http.StatusConflict
		} else {
			return http.StatusInternalServerError
		}
	} else {
		return http.StatusInternalServerError
	}
}
```

Reply:
would you wanna make that change here?

Reply:
oh nice! yep i can make that change